src/api/data/cache/cache.go

```diff
@@ -127,18 +127,18 @@ func (c *cachedCharts) Search(params operations.SearchChartsParams) ([]*models.C
 // Refresh is the interface implementation for data.Charts
 // It refreshes cached data for all authoritative repository+chart data
 func (c *cachedCharts) Refresh() error {
-	c.rwm.Lock()
```

Comment:
The only issue I see with concurrency here is that because all refresh calls share the same pool of workers, it would be possible to negatively impact the performance of the refresh merely by calling the function a whole bunch of times.